src/sampleutil.cpp

```diff
@@ -17,14 +17,62 @@ typedef qint32 int32_t;
 // When changing, be carefull to not prevent the vectorizing
 // https://gcc.gnu.org/projects/tree-ssa/vectorization.html
 
+
+static inline bool useAlignedAlloc() {
+    // Will return true on MSVC 32 bit builds and falls for
+    // Linux 32 and 64 bit builds
+    return (sizeof(long double) == 8 && sizeof(CSAMPLE*) <= 8 &&
+            sizeof(CSAMPLE*) == sizeof(size_t));
+}
+
 // static
 CSAMPLE* SampleUtil::alloc(unsigned int size) {
-    // TODO(XXX) align the array
-    return new CSAMPLE[size];
+    // For optimal use of SSE registers, it is required to align
+    // the sample buffers to 16 Byte (128 bit) boundary
+
+    // Pointers returned by malloc are aligned for the largest scalar type,
+    // which is long double with usually 16 byte.
+    // An exception is MSVC X86 where long double is mapped to double.
+
+    // In case of sizeof(long double) = 8
+    // this code over allocates the requested buffer to be able to
+    // shift the returned pointer to a 16 byte alignment
+    // In the memory before, a pointer to the original
+    // malloced area is stored used to free the memory
+    // This code can be replaced by C11 <stdlib.h> aligned_alloc()
+    // or MSVC ::_aligned_malloc(size, alignment) and  ::_aligned_free(ptr);
+
+    if (useAlignedAlloc()) {
+        // We need to shift the alignment to 16
```

Comment:
There is no need to argue: 
We just need a windows developer who switches this to _aligned_malloc. 
posix_memalign will not help here, because it is already 16 byte aligned on platforms supporting posix_memalign